src/libsync/filesystem.cpp

```diff
@@ -451,22 +451,18 @@ bool FileSystem::setFolderPermissions(const QString &path,
     }
 #endif
 
-#if !defined(Q_OS_MACOS) || __MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_15
     try {
         switch (permissions) {
         case OCC::FileSystem::FolderPermissions::ReadOnly:
             break;
         case OCC::FileSystem::FolderPermissions::ReadWrite:
-            std::filesystem::permissions(path.toStdWString(), std::filesystem::perms::owner_write, std::filesystem::perm_options::add);
+            std::filesystem::permissions(stdStrPath, writePerms, std::filesystem::perm_options::add);
```

Comment:
I am not sure how we should deal with this and users possibly have specific `umask`